src/util/error_collector.go

```diff
@@ -0,0 +1,26 @@
+package util
+
+// FirstError collects errors
+func FirstError(errors ...interface{}) error {
```

Comment:
I'd like to update this interface to 
```go
func FirstError(errorFuncs ...func() error) error {
  //...
}
```

Reply:
This would prevent us from using functions that check errors via side effects, for example by exiting the app if a non-recoverable condition is encountered. An example for those are our numerous `EnsureXXX` functions. 

We could add error return values to those functions, but that would be meaningless and confusing because they always `nil`. The linters would force us to check these errors anyways. Hence, I think overall it is the simplest if the `FirstError` function is generous in what it accepts as input, and tolerates no return value for the `errorFuncs`.

Reply:
Looking at what was changed, I see only a single example of this `prompt.EnsureIsConfigured`. I think its worth adding an extra wrapper around that function (maybe named validateIsConfigured) to simplify this code and let us use full type safety.